runtime/common/RecordLogDecoder.cpp

```diff
@@ -13,118 +13,355 @@ void cudaq::RecordLogDecoder::decode(const std::string &outputLog) {
-
-    switch (currentRecord) {
-    case RecordType::HEADER: {
-      if ("schema_name" == entries[1]) {
-        if ("labeled" == entries[2])
-          schema = SchemaType::LABELED;
-        else if ("ordered" == entries[2])
-          schema = SchemaType::ORDERED;
-        else
-          throw std::runtime_error("Unknown schema type");
-      }
-      /// TODO: Check schema version
-    } break;
-    case RecordType::METADATA:
-      // ignore metadata for now
-      break;
-    case RecordType::START:
-      // indicates start of a shot
-      break;
-    case RecordType::END: {
-      // indicates end of a shot
-      if (entries.size() < 2)
-        throw std::runtime_error("Missing shot status");
-      if ("0" != entries[1])
-        throw std::runtime_error("Cannot handle unsuccessful shot");
-    } break;
-    case RecordType::OUTPUT: {
-      if (entries.size() < 3)
-        throw std::runtime_error("Insufficent data in a record");
-      if ((schema == SchemaType::LABELED) && (entries.size() != 4))
-        throw std::runtime_error("Unexpected record size for a labeled record");
-
-      std::string recType = entries[1];
-      std::string recValue = entries[2];
-      std::string recLabel = (entries.size() == 4) ? entries[3] : "";
-
-      if ("RESULT" == recType)
-        throw std::runtime_error("This type is not yet supported");
-      if ("TUPLE" == recType)
-        throw std::runtime_error("This type is not yet supported");
-      if ("ARRAY" == recType)
-        throw std::runtime_error("This type is not yet supported");
-
-      if ("BOOL" == recType)
-        currentOutput = OutputType::BOOL;
-      else if ("INT" == recType)
-        currentOutput = OutputType::INT;
-      else if ("DOUBLE" == recType)
-        currentOutput = OutputType::DOUBLE;
-      else
-        throw std::runtime_error("Invalid data");
-
-      processSingleRecord(recValue, recLabel);
-    } break;
+      throw std::runtime_error("Unknown schema type");
+  }
+  /// TODO: Handle schema version if needed
+}
+
+void cudaq::RecordLogDecoder::handleMetadata(
+    const std::vector<std::string> &entries) {
+  // Ignore metadata for now
+}
+
+void cudaq::RecordLogDecoder::handleStart(
+    const std::vector<std::string> &entries) {
+  // Ignore start of a shot for now
+}
+
+void cudaq::RecordLogDecoder::handleEnd(
+    const std::vector<std::string> &entries) {
+  if (entries.size() < 2)
+    throw std::runtime_error("Missing shot status");
+  if ("0" != entries[1])
+    throw std::runtime_error("Cannot handle unsuccessful shot");
+}
+
+void cudaq::RecordLogDecoder::handleOutput(
+    const std::vector<std::string> &entries) {
+  if (entries.size() < 3)
+    throw std::runtime_error("Insufficient data in a record");
+  if ((schema == SchemaType::LABELED) && (entries.size() != 4))
+    throw std::runtime_error("Unexpected record size for a labeled record");
+  std::string recType = entries[1];
+  std::string recValue = entries[2];
```

Comment:
Would this change compile? If so, we can save 2 string copies :) 

```suggestion
  const std::string& recType = entries[1];
  const std::string& recValue = entries[2];
```